lib/ansible/modules/cloud/amazon/cloudfront_facts.py

```diff
@@ -0,0 +1,446 @@
+
+    def get_streaming_distribution_config(self, distribution_id):
+        try:
+            func = partial(self.client.get_streaming_distribution_config,Id=distribution_id)
+            return self.paginated_response(func, 'StreamingDistributionConfig')
+        except Exception as e:
+            self.module.fail_json(msg="Error describing streaming distribution - " + str(e), exception=traceback.format_exc(e))
+
+    def list_origin_access_identities(self):
+        try:
+            func = partial(self.client.list_cloud_front_origin_access_identities)
+            return self.paginated_response(func, 'CloudFrontOriginAccessIdentityList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing cloud front origin access identities = " + str(e), exception=traceback.format_exc(e))
+
+    def list_distributions(self):
+        try:
+            func = partial(self.client.list_distributions)
+            return self.paginated_response(func, 'DistributionList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing distributions = " + str(e), exception=traceback.format_exc(e))
+
+    def list_distributions_by_web_acl(self, web_acl_id):
+        try:
+            func = partial(self.client.list_distributions, web_acl_id)
+            return self.paginated_response(func, 'DistributionList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing distributions by web acl id = " + str(e), exception=traceback.format_exc(e))
+
+    def list_invalidations(self):
+        try:
+            func = partial(self.client.list_invalidations)
+            return self.paginated_response(func, 'InvalidationList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing invalidations = " + str(e), exception=traceback.format_exc(e))
+
+    def list_streaming_distributions(self):
+        try:
+            func = partial(self.client.list_streaming_distributions)
+            return self.paginated_response(func, 'StreamingDistributionList')
+        except Exception as e:
+            self.module.fail_json(msg="Error listing streaming distributions = " + str(e), exception=traceback.format_exc(e))
+
+    def get_distribution_id_from_domain_name(self, domain_name):
+        try:
+            distribution_id = ""
+            distributions = self.list_distributions()
+            for dist in distributions['Items']:
+                for alias in dist['Aliases']['Items']:
+                    if str(alias).lower() == domain_name.lower():
+                        distribution_id = str(dist['Id'])
+                        break
+            return distribution_id
+        except Exception as e:
+            self.module.fail_json(msg="Error getting distribution id from domain name = " + str(e), exception=traceback.format_exc(e))
+
+    def paginated_response(self, func, result_key, next_token=None):
```

Comment:
Thanks, @georgepsarakis, your feedback is very welcome! Yes, that is a good idea. Will implement it.